tests/rpm-ostree/main.yml

```diff
@@ -0,0 +1,565 @@
+---
+# vim: set ft=ansible:
+#
+#   This is a general test suite for rpm-ostree that exercises several of the
+# rpm-ostree subcommands.  These tests assume the atomic host is provisioned
+# with the latest version.  These tests are meant to be run in succession.
+#
+- name: rpm-ostree - Setup
+  hosts: all
+  become: yes
+
+  tags:
+    - setup
+
+  vars_files:
+    - vars.yml
+
+  roles:
+    # This playbook requires Ansible 2.2 and an Atomic Host
+    - role: ansible_version_check
+      avc_major: "2"
+      avc_minor: "2"
+      tags:
+        - ansible_version_check
+
+    # Subscribe if the system is RHEL
+    - role: redhat_subscription
+      when: ansible_distribution == 'RedHat'
+      tags:
+        - redhat_subscription
+
+#####################################################################################
+#
+# rpm-ostree deploy version
+#   - validates deploying by version string
+#   - validates cleanup of rollback deployment
+#
+#####################################################################################
+
+- name: rpm-ostree - deploy by version
+  hosts: all
+  become: yes
+
+  tags:
+    - deploy_version
+
+  vars_files:
+    - vars.yml
+
+  tasks:
+    - include: roles/rpm_ostree_status/tasks/main.yml
+
+    - name: Set current version and refspec
+      set_fact:
+         head_version: "{{ ros_booted['version'] }}"
+         refspec: "{{ ros_booted['origin'] }}"
+
+    - name: Pull last two commits
+      command: ostree pull --commit-metadata-only --depth=1 {{ refspec }}
+      register: osp
+      until: osp.rc == 0
+      retries: 5
+      delay: 5
+
+    - name: Get HEAD-1 version
+      shell: ostree show $(ostree rev-parse {{ refspec }}^) --print-metadata-key version | tr -d \'
+      register: hmo_version
+
+    - name: Deploy version {{ hmo_version.stdout }}
+      command: rpm-ostree deploy {{ hmo_version.stdout }}
+      register: ros_deploy
+      until: ros_deploy.rc == 0
+      retries: 5
+      delay: 5
+
+    # reboot
+    - include: ../../common/ans_reboot.yml
+
+    # verify version in deployment 0
+    - include: roles/rpm_ostree_status_verify/tasks/main.yml
```

Comment:
I was kind of all over the place with the include vs include role.  I was trying to use include_role in as many places as possible because it takes out the relative path.  The instances I did not was:

1. ans_reboot.yml isn't in a roles directory so it needs to be included
2. variables not being passed properly using include_role.  I think cevich mentioned something along the lines of this.